### ISSUE TITLE
No results when using cursor and orderBy (Text type)

### ISSUE THREAD

**Issue by redeyecz** [prisma/prisma#6337]
## Bug description

When using findMany I encountered a bug where the logged query from prisma returns data, yet the findMany result does not. I narrowed the bug down to using cursor+orderBy (text column) .... strangely enough, when orderBy is of BigInt type, it seems to work

## How to reproduce

1. Clone this repo https://github.com/redeyecz/prisma-bug
2. docker-compose up
3. http://localhost:3000
4. The shown output prints all database records in the test database, example of cursor+orderBy (bigInt), failed result of cursor+orderBy (text) and expected result

## Expected behavior

I'd expect to get the data from findMany, that the database query returns

## Prisma information

```prisma
model Account {
  id         BigInt   @id @default(autoincrement())
  name       String?
  externalId String?
  type       String
  dateInsert DateTime @default(now())

  @@unique([type, externalId])
  @@map("account")
}
```
```typescript
prisma.account.findMany({
    take: 3,
    skip: 1,
    cursor: {
      id: 9 //Test3
    },
    orderBy: {
      externalId: 'desc' // text column
    }
  })
```

## Environment & setup

Node 14.16

- OS: Windows running WSL2, Docker-engine running Debian 10 (buster) based linux
- Database: Postgres 13.2 (Docker based on alpine image)
- Node.js version: v14.16.0
- Prisma version: 2.20.1

```
prisma               : 2.20.1
@prisma/client       : 2.20.1
Current platform     : debian-openssl-1.1.x
Query Engine         : query-engine 60ba6551f29b17d7d6ce479e5733c70d9c00860e (at node_modules/@prisma/engines/query-engine-debian-openssl-1.1.x)
Migration Engine     : migration-engine-cli 60ba6551f29b17d7d6ce479e5733c70d9c00860e (at node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
Introspection Engine : introspection-core 60ba6551f29b17d7d6ce479e5733c70d9c00860e (at node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x)
Format Binary        : prisma-fmt 60ba6551f29b17d7d6ce479e5733c70d9c00860e (at node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x)
Default Engines Hash : 60ba6551f29b17d7d6ce479e5733c70d9c00860e
Studio               : 0.365.0
```

**Comment by hehex9** [comment]
exactly the same problem, must be a bug in `BigInt`'s implementation, failed to get results with orderBy non-id field.

**Comment by hehex9** [comment]
my minimal and reproducible example: https://github.com/hehex9/prisma-6337

**Comment by pantharshit00** [comment]
Thanks for extremely detailed reproduction @redeyecz and also @hehex9 


I can confirm this. We will patch this.

**Comment by dpetrick** [comment]
A fix will be available in 2.22.
